src/PuliServiceProvider.php

```diff
@@ -55,11 +55,17 @@ public function register(Application $app)
     public function boot(Application $app)
     {
         if (isset($app['twig']) && $app['puli.enable_twig']) {
-            $app['twig.loader'] = $app->share(function (Application $app) {
-                return new PuliTemplateLoader($app['puli.repository']);
+            $app->extend('twig.loader', function ($twigLoader, $app) {
+                $twigLoader->addLoader(new PuliTemplateLoader($app['puli.repository']));
```

Comment:
I would do an instanceof check to see whether you receive an chain loader. If it is not a chain, just create a chain loader with both in it. If it is already a chain loader, just add your own loader (faster than nesting chain loaders into each other)